### PR TITLE
Add list pattern matching to tutorial

### PR DESCRIPTION
**Pull request by rtfeldman**
Closes https://github.com/roc-lang/roc/issues/4452

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -570,6 +570,24 @@ We refer to whatever comes before a `->` in a `when` expression as a *pattern* -
...
+You can also pattern match on lists, like so:
+
+    when myList is
+        [] -> # the list is empty
+        [Foo, ..] -> # it starts with a Foo tag
+        [Foo, Bar, ..] -> # it starts with a Foo tag followed by a Bar tag
+        [Foo, a, ..] -> # it starts with a Foo tag followed by something we name `a`
+        [Ok a, ..] -> # it starts with an Ok tag containing a payload named `a`
+        [.., Foo] -> # it ends with a Foo tag
+        [A, B, .., C, D] -> # it has certain elements at the beginning and and
```
Comment (ayazhafiz) [inline TUTORIAL.md:583]: maybe worth adding a note that this is not valid Roc code, just illustrative?

Comment (ayazhafiz) [reply TUTORIAL.md:583]: Consider also including the following patterns:

```
[_, ..] -> # the list starts with an element, which we ignore
[A, B, C] -> # the list contains exactly three elements, A, B, C
```

Comment (rtfeldman) [reply TUTORIAL.md]: Cool! Updated in 99b48c31628d123392e93ccfabb8fa4ce7a86b1e - how's that look?

**Review by ayazhafiz — APPROVED** [review]
😍

**Pull request merged**